bakta_queries_unittest.py

```diff
@@ -0,0 +1,118 @@
+        self.assertIsInstance(cursor, sqlite3.Cursor)
+
+    def test_close_database_connection(self):
+        conn, _ = connect_to_database(self.db_connection)  # Fixed this line
+        close_database_connection(conn)
+        with self.assertRaises(sqlite3.ProgrammingError):
+            conn.execute("SELECT 1")
+
+    def create_test_tables(self):
+        # Create the 'bakta' and 'identifier' tables for testing
+        self.db_cursor.execute(
+            """
+            CREATE TABLE IF NOT EXISTS bakta (
+                contig_id TEXT,
+                start INT,
+                end INT,
+                strand TEXT,
+                entity_id TEXT,
+                gene_id TEXT,
+                filename TEXT,
+                filepath TEXT
+            )
+            """
+        )
+
+        self.db_cursor.execute(
+            """
+            CREATE TABLE IF NOT EXISTS identifier (
+                entity_id TEXT,
+                filename TEXT,
+                filepath TEXT
+            )
+            """
+        )
+
+        # Insert test data into 'bakta' table
+        self.db_cursor.execute(
+            """
+            INSERT INTO bakta VALUES (?, ?, ?, ?, ?, ?, ?, ?)
+            """,
+            ("Contig1", 1, 10, "1", "Entity1", "Gene1", "File1", "/path/to/files"),
+        )
+
+    def test_get_gene_info_by_name(self):
+        conn, cursor = connect_to_database(self.db_connection)  # Fixed this line
+        # Replace with your test data
+        test_gene_name = "Gene1"
+        test_column = "gene_id"
+        cursor.execute(
+            "CREATE TABLE IF NOT EXISTS bakta (contig_id TEXT, start INT, end INT, strand TEXT, entity_id TEXT, gene_id TEXT, filename TEXT, filepath TEXT)"
+        )
+        cursor.execute(
+            "CREATE TABLE IF NOT EXISTS identifier (entity_id TEXT, filename TEXT, filepath TEXT)"
+        )
+        cursor.execute(
+            "INSERT INTO bakta VALUES (?, ?, ?, ?, ?, ?, ?, ?)",
+            ("Contig1", 1, 10, "1", "Entity1", "Gene1", "File1", "/path/to/files"),
+        )
+        gene_info = get_gene_info_by_name(cursor, test_gene_name, test_column)
+        print(gene_info)
+        self.assertEqual(len(gene_info), 1)
+        conn.close()
+
+    def test_extract_gene_sequence_from_fasta(self):
+        fasta_file_path = os.path.join(self.temp_dir.name, "test.fasta")
+        # Create a test FASTA file
+        with open(fasta_file_path, "w") as fasta_file:
+            fasta_file.write(">Contig1\nACGT")
+
+        gene_sequence = extract_gene_sequence_from_fasta(fasta_file_path, "Contig1", 1, 4, "1")
+        self.assertEqual(gene_sequence, "CGT")
+
+    def test_write_gene_sequences_to_fasta(self):
+        output_file_path = os.path.join(self.temp_dir.name, "test_output.fasta")
+        gene_sequences = ">Gene1\nACGT\n>Gene2\nTGCA\n"
+        write_gene_sequences_to_fasta(output_file_path, gene_sequences)
+        self.assertTrue(os.path.isfile(output_file_path))
```

Comment:
Python has extremely helpful IO mechanisms for stuff like this, consider:

```python
    import io
    outfile = io.StringIO()
    write_gene_sequences_to_fasta(outfile, gene_sequences)
    exp = '\n'.join(['>foo', 'ATGC', 
                     '>bar', 'TGCA', ''])  # empty string at the end to get an ending newline
    outfile.seek(0)
    obs = outfile.read()
    self.assertEqual(obs, exp)
```

The above approach doesn't involved monkeying with temp directories, and explicitly tests the content